src/main/java/org/folio/service/inventory/InventoryUtils.java

```diff
@@ -272,4 +274,17 @@ public static void updateItemWithReceivedItemFields(JsonObject item, ReceivedIte
     }
   }
 
+  public static ItemRecreateConfig constructItemRecreateConfig(String receivingTenantId, RequestContext requestContext, boolean reuseInitialRequestContext) {
+    if (Objects.isNull(receivingTenantId)) {
+      return new ItemRecreateConfig(null, reuseInitialRequestContext ? requestContext : null);
+    }
+    return new ItemRecreateConfig(receivingTenantId, createContextWithNewTenantId(requestContext, receivingTenantId));
+  }
+
+  public static boolean allowItemRecreate(ItemRecreateConfig srcConfig, ItemRecreateConfig dstConfig) {
+    return Objects.nonNull(srcConfig.tenantId()) && Objects.nonNull(dstConfig.tenantId())
+      && !srcConfig.tenantId().equals(dstConfig.tenantId());
+  }
+
+  public record ItemRecreateConfig(String tenantId, RequestContext context) {}
```

Comment:
Correct me if I am wrong, but I think RequestContext already contains the tenant id which can be extracted. In other places different contexts are used to manage cross-tenant actions. I see these config records are only used for `allowItemRecreate` so maybe this could be cleaned up and this method could be modified to accept request contexts. What do you think?

Reply:
> Correct me if I am wrong, but I think RequestContext already contains the tenant id which can be extracted. In other places different contexts are used to manage cross-tenant actions. I see these config records are only used for `allowItemRecreate` so maybe this could be cleaned up and this method could be modified to accept request contexts. What do you think?

No, this Record and its RequestContext is used to move item(s) from one tenant to the other, and not just for validation. It is  used downstream in context switching in both Quick and Fullscreen Receive.